### ISSUE TITLE
Migrated ephemeral_disk data does not retain ownership for directories

### ISSUE THREAD

**Issue by pam9999** [hashicorp/nomad#3702]
If filing a bug please include the following:

### Nomad version
0.7.0

### Operating system and Environment details
RHEL 7.4 with umask 0027
### Issue
We use a MariaDB docker container and MariaDB create one directory for each database. 
When the allocation directory is migrated from one nomad client to another, ownership on these directories are not retained. Directories are created as root and the Docker process (running with mysql user) cannot access anymore to the databases directories.
### Reproduction steps
Create a job spec with a mariadb task in a docker container (i.e. Redhat rhscl/mariadb-101-rhel7 image).
Run the job and create a database (directories are created with mysql ownership)
Drain the node
Directories are created with root ownership and docker process cannot access these directories anymore because of the restrictive umask.

We patched  alloc_watcher.go

```go
        // If the header is for a directory we create the directory
        if hdr.Typeflag == tar.TypeDir {
            os.MkdirAll(filepath.Join(dest, hdr.Name), os.FileMode(hdr.Mode))
            f, err := os.Open(filepath.Join(dest, hdr.Name))
            if err != nil {
                return fmt.Errorf("error open directory: %v", err)
            }
            if err := f.Chown(hdr.Uid, hdr.Gid); err != nil {
                f.Close()
                return fmt.Errorf("error chowning directory %v", err)
            }
            continue
        }
```

**Comment by chelseakomlo** [comment]
Thanks for submitting this issue- we'll reproduce it and test this patch as well.

**Comment by schmichael** [comment]
Fixed in #3723 which has some binaries with the fix attached if you want to test. Thanks again @pam9999!
